hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
Right, but the following works:
```sh
$ cat working-array.sh
#!/bin/bash
set -o nounset
tests_packages=()
tests_packages+=('0th element')
echo "${tests_packages[0]}"
$ ./working-array.sh
0th element
```

You can furthermore get the length (`"${#array[@]}"`) without getting an unbound error. I think it's a lot cleaner to check the length, don't access the 0-length array, then keeping a null value in the first slot. 

Or, even better, check if there are non-flag arguments passed in. We never expect `test_packages` to be empty, so if it is or if there aren't any package arguments passed in we can raise an error there.